spyder/plugins/ipythonconsole/widgets/client.py

```diff
@@ -199,11 +176,51 @@ def __del__(self):
             self.restart_thread.quit()
             self.restart_thread.wait()
 
+    @property
+    def connection_file(self):
+        if self.kernel is None:
+            return None
+        return self.kernel.connection_file
+
+    @property
+    def stderr_obj(self):
+        if self.kernel is None:
+            return None
+        return self.kernel.stderr_obj
+
+    @property
+    def stdout_obj(self):
+        if self.kernel is None:
+            return None
+        return self.kernel.stdout_obj
+
+    @property
+    def fault_obj(self):
+        if self.kernel is None:
+            return None
+        return self.kernel.fault_obj
+
+    def start_std_poll(self):
+        """Start polling std files"""
+        self.std_poll_timer = QTimer(self)
+        self.std_poll_timer.timeout.connect(self.poll_std_file_change)
+        self.std_poll_timer.setInterval(1000)
+        self.std_poll_timer.start()
+        self.shellwidget.executed.connect(self.poll_std_file_change)
+
+    def connect_kernel(self, kernel):
+        """Connect kernel to client."""
+        self._before_prompt_is_ready()
+        self.kernel = kernel
+        if kernel.stderr_obj is not None or kernel.stdout_obj is not None:
+            self.start_std_poll()
+        # Actually do the connection
+        self.shellwidget.connect_kernel(kernel)
```

Comment:
Please move these methods to an appropriate comment section or sections below (e.g. `Public API`) or create new sections if necessary.

For the future, please take into account that it's increasingly difficult to contribute to the IPython console if we keep adding up methods here and there without a clear structure.